bindings/ocaml/tests/aead_test.ml

```diff
@@ -103,10 +156,20 @@ let _ =
   Printf.printf "MOVBE: %b\n" (has_feature MOVBE);
   Printf.printf "RDRAND: %b\n" (has_feature RDRAND);
 
+  test_agile_noalloc chacha20poly1305_test;
   test_agile chacha20poly1305_test;
-  test_nonagile chacha20poly1305_test "Hacl.Chacha20_Poly1305_32" Hacl.Chacha20_Poly1305_32.encrypt Hacl.Chacha20_Poly1305_32.decrypt [];
-  test_nonagile chacha20poly1305_test "Hacl.Chacha20_Poly1305_128" Hacl.Chacha20_Poly1305_128.encrypt Hacl.Chacha20_Poly1305_128.decrypt [VEC128];
-  test_nonagile chacha20poly1305_test "Hacl.Chacha20_Poly1305_256" Hacl.Chacha20_Poly1305_256.encrypt Hacl.Chacha20_Poly1305_256.decrypt [VEC256];
-  test_nonagile chacha20poly1305_test "EverCrypt.Chacha20_Poly1305_256" EverCrypt.Chacha20_Poly1305.encrypt EverCrypt.Chacha20_Poly1305.decrypt [];
 
+  let module Tests = MakeTests (EverCrypt.Chacha20_Poly1305) in
```

Comment:
Using first-class modules here, maybe this breaks backwards-compatibility with really old versions of OCaml ?

Reply:
Ah didn't think of that. Seems they were introduced in OCaml 3.12, so I don't think we care too much? (We certainly don't at Nomadic). In any case, they've been in different parts of the codebase from the beginning, like in `SharedFunctors.ml`.

Reply:
Wouahou I didn't think they were so old, OK I don't think we need to maintain compat below 3.12.

Reply:
Clearly, first-class modules are old enough and this is definitely a perfect use-case for them. Does that make you an old-timer Denis?